### PR TITLE
Re-add hook for LivingExperienceDropEvent, fixes #7900

### LINKED ISSUE #7900: [1.17.1] LivingExperienceDropEvent is not fired (no event injected)
**Minecraft Version:** 1.17.1

**Forge Version:** 37.0.1

**Logs:** N/A

**Steps to Reproduce:**
 1. Utilize the LivingExperienceDropEvent
 2. It doesn't get fired

**Description of issue:**

LivingExperienceDropEvent is not fired at all. Analyzing Forge's code, in 1.16 you can trace back from the event to ForgeEventFactory#getExperienceDrop, and find usages from there to bring you to LivingEntity#dropExperience wherein you can find the method being called. In 1.17, this method call is missing entirely.

### PR DESCRIPTION
**Pull request by AterAnimAvis**
Re-adds the hook that got missed during the 1.17 port.

Closes #7900

Testing:
----- 
- Enable Test Mod
- Join World
- Spawn a `pig` and kill it
- Verify the logged message
- Verify via `/xp query points` that the received xp was double the reported original.

### CODE REVIEW

File: src/test/java/net/minecraftforge/debug/entity/living/LivingExperienceDropEventTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package net.minecraftforge.debug.entity.living;
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/entity/living/LivingExperienceDropEventTest.java:20]: Missing license header.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Looks good to me, except for a missing license header.

File: src/test/java/net/minecraftforge/debug/entity/living/LivingExperienceDropEventTest.java (hunk 1 above)
Comment (AterAnimAvis) [reply src/test/java/net/minecraftforge/debug/entity/living/LivingExperienceDropEventTest.java:20]: Whoops

**Review by sciwhiz12 — APPROVED** [review]

**Comment by LexManos** [comment]
Dragons

**Comment by AterAnimAvis** [comment]
Closing this for the moment with the aim to retarget to 1.18, EnderDragons are more complex to account for (The attacking player may be null at the point of time the event fires)

**Pull request closed without merging**